src/experimental/tools/download2go/apis/dash/dashConnectivity.ts

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright 2019 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Observable, of, throwError } from "rxjs";
+import { mergeMap } from "rxjs/operators";
+
+import { ISegment } from "../../../../../manifest";
+import { ISidxSegment } from "../../../../../parsers/containers/isobmff";
+import parseManifest from "../../../../../parsers/manifest/dash/index";
+import { IParserResponse } from "../../../../../parsers/manifest/dash/parse_mpd";
+import { IParsedManifest } from "../../../../../parsers/manifest/types";
+import xhrRequest from "../../../../../utils/request";
+import { makeHTTPRequest, SegmentConstuctionError } from "../../utils";
+import { ITypedArray } from "../drm/keySystems";
+import { IUtils } from "./../../types";
+import {
+  IRepresentation,
+  ISegmentBuilder,
+  ISegmentBuilt,
+  ISegmentsBuiltType,
+} from "./types";
+
+/**
+ * Get a manifest from a server and parse it.
+ *
+ * @param url - The url manifest mpd file
+ * @returns The manifest parsed IParserResponse<IParsedManifest>
+ *
+ */
+export const getOnlineManifest = (url: string) => {
+  return xhrRequest({
+    url,
+    responseType: "text",
+  }).pipe(
+    mergeMap(({ value }) =>
```

Comment:
In that case you could simply use a `map` operator and return the result from `parseManifest`, no ?